postpic/particles/particles.py

```diff
@@ -1279,3 +1279,38 @@ def collect(self, *scalarfs):
                 particlelist[i].append(scalars[:, k])
         ret = [np.asarray(p).T for p in particlelist]
         return ret
+
+    def exportToHdf5(self, filename, *scalarfs):
+        '''
+        Exports the collected data to an hdf5-file. If the file with the same name
+        already exists, it will be deleted!
+        Parameters:
+        -----------
+        filename:   whole name or path of the file which the data will be exported to.
+        *scalarfs:  the scalarfunction(s) defining the particle property.
+                Id and weight will always be stored, so they do not have to be
+                specified here.
+
+        Returns:
+        -----------
+        The same as collect(): The list of particles with the defined properties.
+        The hdf5 file will have the structure:
+        - dset: Ids
+        - particles/Id/track: track data (ID, weight, scalarfs)
+        - particles/Id/weight: weights
+        '''
+        scalarfs = ("id", "weight") + scalarfs
+        part_data = self.collect(*scalarfs)
+
+        import h5py
+
+        f = h5py.File(filename, "w")
+        dset_ids = f.create_dataset("Ids", data=self.ids, dtype=np.int64)
+
+        for id in range(0, len(self.ids)):
+            f.create_dataset("particles/{}/track".format(int(self.ids[id])), data=part_data[id])
```

Comment:
To make hdf5 self describing, we can add the string of the data, which is written to the track file. 
I think it should be something like
`f.attrs['scalars'] = scalarfs`
but please check if you can access this in the final hdf5.